contract/tests/test_contract.py

```diff
@@ -317,7 +321,9 @@ def test_contract_invoice_followers(self):
         self.contract.message_subscribe(
             partner_ids=self.contract.partner_id.ids, subtype_ids=subtype_ids
         )
-        self.contract._recurring_create_invoice()
+        self.env["contract.manually.single.invoice"].create(
```

Comment:
I don't think you should change this test, but only add a new one for the new use case. This also proves that there's no change in the previous behavior for compatibility reasons.